syntax = "proto3";

message RequisicaoRegistro {
   string servico = 1;
   repeated uint64 listaChaves = 2;
}

message RespostaRegistro {
  uint64 chavesProcessadas = 1;
}

message RequisicaoMapeamento {
  uint64 chave = 1;
}

message RespostaMapeamento {
  string servico = 1;
}

message RequisicaoTermino {}

message RespostaTerminoCentralizador {
  uint64 chavesRegistradas = 1;
}

message RespostaTerminoPares {
  uint64 status = 1;
}

message RequisicaoInsercao {
  uint64 chave = 1;
  string valor = 2;
}

message RespostaInsercao {
  uint64 status = 1;
}

message RequisicaoConsulta {
  uint64 chave = 1;
}

message RespostaConsulta {
  string resultado = 1;
}

message RequisicaoAtivacao {
  string servico = 1;
}

message RespostaAtivacao {
  uint64 chavesProcessadas = 1;
}


service ServerCentralizador {
  rpc Registrar(RequisicaoRegistro) returns(RespostaRegistro) {}
  rpc Mapear(RequisicaoMapeamento) returns(RespostaMapeamento) {}
  rpc Termina(RequisicaoTermino) returns(RespostaTerminoCentralizador) {}
}

service ServerPares {
  rpc Insere(RequisicaoInsercao) returns(RespostaInsercao) {}
  rpc Consulta(RequisicaoConsulta) returns(RespostaConsulta) {}
  rpc Ativa(RequisicaoAtivacao) returns(RespostaAtivacao) {}
  rpc Termina(RequisicaoTermino) returns(RespostaTerminoPares) {}
}


